packages/framework/src/Component/Grid/ArrayWithPaginationDataSource.php

```diff
@@ -12,10 +12,10 @@ class ArrayWithPaginationDataSource extends ArrayDataSource
      * {@inheritdoc}
      */
     public function getPaginatedRows(
-        $limit = null,
-        $page = 1,
-        $orderSourceColumnName = null,
-        $orderDirection = self::ORDER_ASC,
+        ?int $limit = null,
+        int $page = 1,
+        ?string $orderSourceColumnName = null,
+        ?string $orderDirection = self::ORDER_ASC,
```

Comment:
Is it necessary to have the `orderDirection` nullable? What ordering type means `null`?

In the end, the value is passed to for example `QueryBuilderDataSource::addQueryOrder`, where the `orderDirection` is typed as a string, so it may fail on error

(several occurrences in this PR)

Reply:
I have chaged it